Library/Homebrew/utils/curl.rb

```diff
@@ -271,7 +281,7 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default],
       file_match = details[:file_hash] == secure_details[:file_hash]
 
       if (etag_match || content_length_match || file_match) &&
-         secure_details[:final_url].start_with?("https://") &&
+         secure_details[:final_url]&.start_with?("https://") &&
```

Comment:
```suggestion
         secure_details[:final_url].present? && secure_details[:final_url].start_with?("https://")
```

Reply:
Since this compound condition is used in two places (with the only difference being that one starts with `url.start_with?("http://")` and the other ends with it), I've also split this out into a `http_with_https_available` variable. This hopefully makes the related conditions easier to read while avoiding some code duplication.